### PR TITLE
ci: Add pre-commit.ci configuration

### PR DESCRIPTION
**Pull request by nijel**
## Proposed changes

It can be extended in the future to include more code formatting tools. This is very minimal configuration to introduce the service. 

To use [pre-commit.ci](https://pre-commit.ci/) the application access has to be approved by @python-social-auth admin (@omab), I've already sent request for that.

## Types of changes

Please check the type of change your PR introduces:

- [ ] Release (new release request)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (PEP8, lint, formatting, renaming, etc)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Build related changes (build process, tests runner, etc)
- [ ] Other (please describe):

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask. We're here to
help! This is simply a reminder of what we are going to look for before merging
your code._

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

## Other information

Any other information that is important to this PR such as screenshots of how
the component looks before and after the change.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
```
Comment (atodorov) [inline .pre-commit-config.yaml:5]: LGTM. I am only concerned about having to manually upgrade the versions of these pre-commit hooks. 

I know Dependabot can upgrade GitHub actions but it looks like it doesn't support pre-commit hooks.

**Review by atodorov — APPROVED** [review]
LGTM

File: .pre-commit-config.yaml (hunk 1 above)
Comment (nijel) [reply .pre-commit-config.yaml:5]: The service does upgrades weekly, for example see https://github.com/asottile/pyupgrade/pull/570

**Review by digismack — APPROVED** [review]

**Comment by nijel** [comment]
To make this useful, we still need approval of [pre-commit.ci app](https://github.com/apps/pre-commit-ci) on @python-social-auth organization.

**Comment by nijel** [comment]
@omab Can you please allow pre-commit ci at https://github.com/organizations/python-social-auth/settings/installations ?

**Pull request merged**